File: Math/382-increasingtripleS.py

#LC 334
from cmath import inf


def isIncreasingTriples(nums):
    
    a,b=inf,inf
    for i in nums:
        if i<=a:
            a=i
        elif i<=b:
            b=i
        else:
            return True
    return False


nums=[1,0,-1,2,0,4,5]
print(isIncreasingTriples(nums))

File: Math/382-increasingtripleS.cpp
#include<vector>
#include<iostream>

using VEC=std::vector<int>;

bool isIncreasingTriples(VEC& nums)
{
    int a=INT_MAX;
    int b=INT_MAX;
    for(auto &v:nums)
    {
        if(v<=a)
            a=v;
        else if(v<=b)
            b=v;
        else
            return true;
    }
    return false;
}

int main()
{
    VEC nums{1,0,-1,2,0,4,5};
    std::cout<<isIncreasingTriples(nums);
    return 0;
}